lib/features/palette/PaletteProvider.js

```diff
@@ -32,7 +36,9 @@ PaletteProvider.$inject = [
   'lassoTool',
   'handTool',
   'globalConnect',
-  'translate'
+  'translate',
+  'popupMenu',
```

Comment:
One general question that I had is "should we make create/append anything" available as a pluggable module => You can either have it or not.

This question comes up any time I see us hooking into existing behaviors, where we could just build the feature self-contained.

Why could that be beneficial: Because users then have the choice to benefit from our modular architecture, either use it, or not.

(Same could be said for other things of course, such as aligmnent utilities, ...)

Reply:
> [not include these in the base modeler](https://github.com/bpmn-io/bpmn-js/blob/11a7666ff111d84c9d19f400c71cf4fed54e779b/lib/Modeler.js#L180)

We can include it in the base modeler, as there exists the ["really base modeler"](https://github.com/bpmn-io/bpmn-js/blob/11a7666ff111d84c9d19f400c71cf4fed54e779b/lib/BaseModeler.js#L180) people can use to customize (pick only what they actually need).

Reply:
Consider combining both in a shared place to simplify things. "create/append anything" is a feature.